File: Paradigmas_de_Programacao/Atividades/Atividade_5/11.hs
primeiros :: Int -> [t] -> [t]
primeiros 0 l = []
primeiros n [] = []
primeiros n (a:b) = a : primeiros (n-1) b

main = do
    print(primeiros 3 [1, 2, 3, 4, 5, 6])

File: INE5416/Atividades/Atividade_16/11.pl
:- initialization(main).

primeiros(_, [], []).
primeiros(0, _, []).
primeiros(1, [H|_], [H]).
primeiros(N, [H|T1], [H|T2]) :- N1 is N-1, primeiros(N1, T1, T2).

main :- primeiros(3, [1, 2, 3, 4, 5], R),
        writeln(R),
        halt.
